menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -57,16 +63,35 @@ class SchemaController @Autowired() (
     val fileContent = new String(file.getBytes)
 
     val sparkStruct = origFormat match {
-      case Some("struct") | Some("") | None => sparkMenasConvertor.convertAnyToStructType(fileContent)
-      case Some("copybook") =>
-        val parsedSchema = CopybookParser.parseTree(fileContent)
-        val cobolSchema = new CobolSchema(parsedSchema, SchemaRetentionPolicy.CollapseRoot, false)
-        cobolSchema.getSparkSchema
-      case Some(x) => throw new UnsupportedOperationException(s"$x is not a recognized schema format. Menas currently supports: 'struct' and 'copybook'")
+      case Some(SchemaTypeStruct) | Some("") | None =>
+        try {
+          sparkMenasConvertor.convertAnyToStructType(fileContent)
+        } catch {
+          case NonFatal(e) => throw SchemaParsingException(SchemaTypeStruct, e.getMessage)
+        }
+      case Some(SchemaTypeCopybook) =>
+        try {
+          val parsedSchema = CopybookParser.parseTree(fileContent)
+          val cobolSchema = new CobolSchema(parsedSchema, SchemaRetentionPolicy.CollapseRoot, false)
+          cobolSchema.getSparkSchema
+        } catch {
+          case e: SyntaxErrorException =>
+            throw SchemaParsingException(SchemaTypeCopybook, e.getMessage, Some(e.lineNumber), None, Some(e.field), e)
+          case NonFatal(e) =>
+            throw SchemaParsingException(SchemaTypeCopybook, e.getMessage)
+        }
+      case Some(schemaType) =>
+        throw SchemaFormatException(schemaType, s"$schemaType is not a recognized schema format. " +
+          s"Menas currently supports: '$SchemaTypeStruct' and '$SchemaTypeCopybook'.")
```

Comment:
```suggestion
          s"Menas currently supports: '$SchemaTypeStruct' and '$SchemaTypeCopybook'.", e)
```
Let's add the cause here.

Reply:
This is an initial exception.

Reply:
lol, I was on a roll from the other ones